pkg/controllers/clusterset/clusterrole/rbac.go

```diff
@@ -0,0 +1,66 @@
+// licensed Materials - Property of IBM
+// 5737-E67
+// (C) Copyright IBM Corporation 2016, 2019 All Rights Reserved
+// US Government Users Restricted Rights - Use, duplication or disclosure restricted by GSA ADP Schedule Contract with IBM Corp.
+
+package clusterrole
+
+import (
+	"github.com/open-cluster-management/multicloud-operators-foundation/pkg/controllers/clusterrbac"
+	rbacv1 "k8s.io/api/rbac/v1"
+)
+
+var managedclusterGroup = "cluster.open-cluster-management.io"
+var hiveGroup = "hive.openshift.io"
+
+// buildAdminRoleRules builds the clustesetadminroles
+func buildAdminRoleRules(clustersetName string) []rbacv1.PolicyRule {
+	return []rbacv1.PolicyRule{
+		clusterrbac.NewRule("get", "update").
+			Groups(managedclusterGroup).
+			Resources("managedclustersets").
+			Names(clustersetName).
+			RuleOrDie(),
+		clusterrbac.NewRule("create").
+			Groups(managedclusterGroup).
+			Resources("managedclustersets/join").
+			Names(clustersetName).
+			RuleOrDie(),
+		clusterrbac.NewRule("create").
+			Groups(managedclusterGroup).
+			Resources("managedclustersets/bind").
+			Names(clustersetName).
+			RuleOrDie(),
+		clusterrbac.NewRule("create", "get", "list", "watch", "update", "delete").
```

Comment:
please add some doc here. I think we add the following rules because we want clusterset admin to be  able to provision clusters.